solutions/USACO open20-social-distancing.mdx

```diff
@@ -11,6 +11,76 @@ author: Qi Wang
 **Time Complexity:** $\mathcal{O}((N+M)\log (maxDist))$
 
 <LanguageSection>
+    
+<CPPSection>
+
+## C++ Implementation 
+
+```cpp
+#include <bits/stdc++.h>
+    
+using namespace std;
+
+#define ll long long
+
+int n, m;
+pair<ll,ll> intervals[100005];
+void setIO(string name, bool includeout=false) { // name is nonempty for USACO file I/O
+    ios_base::sync_with_stdio(0); cin.tie(0); // see Fast Input & Output
+    // alternatively, cin.tie(0)->sync_with_stdio(0);
+    freopen((name+".in").c_str(), "r", stdin); // see Input & Output
+    if (includeout) {
+        freopen((name+".out").c_str(), "w", stdout);
+    }
+}
+bool check(ll dist) {
+    ll currPoint = intervals[0].first; //starting off at earliest possible point
+    ll currInterval = 0; //what patch we are currently on
+    ll placed = 0; //how many cows have we placed in total
+    while (placed < n) {
+        if (intervals[currInterval].first <= currPoint && 
+        intervals[currInterval].second >= currPoint) {
```

Comment:
```suggestion
            intervals[currInterval].second >= currPoint) {
```
oops

Reply:
edit: 8 spaces instead of 4... also why does the `if` statement and the second condition line up?

Reply:
im so confused